update Implementation Packages docs (#435)
digital-asset/daml-finance: docs/source/packages/implementations/daml-finance-claims.rst

This package contains utility functions that facilitate building and working with
contingent-claim-based instruments. It includes the following modules:

- :ref:`Lifecycle.Rule <module-daml-finance-claims-lifecycle-rule-53980>`:

>   Rule to process a time update event for instruments that are modelled using "on-the-fly" claims

Reviewer: ```suggestion
  Rule to process a time update event for instruments that are modelled using a dynamic claim representation
```

? I think we should choose a terminology to use in the open world for what we refer to as "on-the-fly" :)

Author: I have improved it by linking to a page that explains what we mean by on-the-fly. The expression itself is quite widely used, so I think we just need to specify what we mean by it for claims.
